storage/txcache/txCacheFailsafe.go

```diff
@@ -0,0 +1,140 @@
+}
+
+func newTxCacheFailsafe(cache txCache) txCache {
+	return &txCacheFailsafeDecorator{
+		backingCache: cache,
+	}
+}
+
+// AddTx delegates to backing cache, with try / catch
+func (decorator *txCacheFailsafeDecorator) AddTx(tx *WrappedTransaction) (ok bool, added bool) {
+	core.TryCatch(func() {
+		ok, added = decorator.backingCache.AddTx(tx)
+	}, decorator.catchAndLog, "AddTx")
+
+	return
+}
+
+// GetByTxHash delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) GetByTxHash(txHash []byte) (*WrappedTransaction, bool) {
+	return decorator.backingCache.GetByTxHash(txHash)
+}
+
+// SelectTransactions deledates to backing cache, with try / cache
+func (decorator *txCacheFailsafeDecorator) SelectTransactions(numRequested int, batchSizePerSender int) []*WrappedTransaction {
+	selection := make([]*WrappedTransaction, 0)
+
+	core.TryCatch(func() {
+		selection = decorator.backingCache.SelectTransactions(numRequested, batchSizePerSender)
+	}, decorator.catchAndLog, "SelectTransactions")
+
+	return selection
+}
+
+// RemoveTxByHash delegates to backing cache, with try / cache
+func (decorator *txCacheFailsafeDecorator) RemoveTxByHash(txHash []byte) error {
+	var err error
+
+	core.TryCatch(func() {
+		err = decorator.backingCache.RemoveTxByHash(txHash)
+	}, decorator.catchAndLog, "RemoveTxByHash")
+
+	return err
+}
+
+// CountTx delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) CountTx() uint64 {
+	return decorator.backingCache.CountTx()
+}
+
+// Len delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) Len() int {
+	return decorator.backingCache.Len()
+}
+
+// ForEachTransaction delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) ForEachTransaction(function ForEachTransaction) {
+	decorator.backingCache.ForEachTransaction(function)
+}
+
+// Clear delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) Clear() {
+	decorator.backingCache.Clear()
+}
+
+// Put delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) Put(key []byte, value interface{}) (evicted bool) {
+	return decorator.backingCache.Put(key, value)
+}
+
+// Get delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) Get(key []byte) (value interface{}, ok bool) {
+	return decorator.backingCache.Get(key)
+}
+
+// Has delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) Has(key []byte) bool {
+	return decorator.backingCache.Has(key)
+}
+
+// Peek delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) Peek(key []byte) (value interface{}, ok bool) {
+	return decorator.backingCache.Peek(key)
+}
+
+// HasOrAdd delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) HasOrAdd(key []byte, value interface{}) (ok, evicted bool) {
+	return decorator.backingCache.HasOrAdd(key, value)
+}
+
+// Remove delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) Remove(key []byte) {
+	decorator.RemoveTxByHash(key)
```

Comment:
_= unhandled error

Reply:
The _failSafe_ component was removed in the fix-after-review.